google_guest_agent/hostname/hostname_windows_test.go

```diff
@@ -0,0 +1,31 @@
+//go:build windows
+
+package hostname
+
+import (
+	"syscall"
+	"testing"
+	"time"
+)
+
+func TestNotifyIpInterfaceChange(t *testing.T) {
+	var handle uintptr
+	var callbackExecuted bool
+	callback := func() uintptr {
+		callbackExecuted = true
+		return 0
+	}
+	if err := notifyIpInterfaceChange(syscall.AF_UNSPEC, syscall.NewCallback(callback), nil, true, &handle); err != nil {
+		t.Errorf("failed to register callback: %v", err)
+	}
+	if handle == 0 {
+		t.Error("notification handle is nil after registering callback")
+	}
+	time.Sleep(1 * time.Second)
```

Comment:
if you really need to wait until configs are reflected you better follow retry strategy (see retry package).

Reply:
It wasn't for configs to be reflected, it was for the callback to execute. I thought that at least `sleep(0)` would be needed to give up the thread execution but I was wrong, it passed with `--test.count 200` after I removed it.